Changed structure to separate presentations from discussion summaries… (#2)
nimh-dsst/lunch-and-learn: discussions/IDEs.md

> # Interactive Development Environments (IDEs) Discussion

Reviewer: ```suggestion
# Integrated Development Environments (IDEs) Discussion
```

> ## IDEs used by Data Science and Sharing Team and Friends

Reviewer: ```suggestion
## IDEs used by Data Science & Sharing Team and Friends
```

> * [VS Code](https://code.visualstudio.com/)

Reviewer: ```suggestion
* [Visual Studio Code (a.k.a. V.S. Code)](https://code.visualstudio.com/)
```

During the discussions users brought up pros and cons of various IDEs. Here are some of the highlights:

### Git Integration

Most IDEs offer GUI interfaces for common git actions. However, some users prefered using git from the command line instead.

### VS Code Considerations

* VS Code's [testing interface](https://code.visualstudio.com/docs/editor/testing) provides support for Python's `unittest` and `pytest`.
* Extensions in VS Code
  * [Python Extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python) by Microsoft
  * [GitHub Pull Requests Extension](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github)
  * [Black Formatter](https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter)
  * [Flake8](https://marketplace.visualstudio.com/items?itemName=ms-python.flake8)
  * [isort](https://marketplace.visualstudio.com/items?itemName=ms-python.isort)
  * [GitHub Copilot](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot). Requires a paid subscription.
  * [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat). Requires a paid subscription.

> 

Reviewer: ```suggestion
  * [Rainbow CSV](https://marketplace.visualstudio.com/items?itemName=mechatroner.rainbow-csv) for all your something-separated-value file needs.
  * [Remote SSH](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) for developing over SSH on different machines.
  
```

### Do you need to see the data?

> An interesting point of discusssion was raised about whether or not users preferred to see the data. Spyder comes with a variable explorer panel. Jupyter notebooks facilitate pandas DataFrame visualizations and plotting. For Data Science and Analytics tasks, sometime a faciliated view into the data can be helpful. Sometimes it is helpful to be able to traverse tabular data in Excel.

Reviewer: ```suggestion
An interesting point of discusssion was raised about whether or not users preferred to see the data. Spyder comes with a variable explorer panel. Jupyter notebooks facilitate pandas DataFrame visualizations and plotting. For Data Science and Analytics tasks, sometimes a facilitated view into the data can be helpful. Sometimes it is helpful to be able to traverse tabular data in Excel.
```
